src/sparse/impl/KokkosSparse_spgemm_mkl_impl.hpp

```diff
@@ -538,7 +538,13 @@ void mkl_symbolic(
 
 
           //KokkosKernels::Impl::copy_vector<MKL_INT *, typename cin_row_index_view_type::non_const_type, MyExecSpace> (m, rows_start, row_mapC);
-          idx nnz = row_mapC(m) =  rows_end[m - 1];
+          //idx nnz = row_mapC(m) = rows_end[m - 1];
+          idx nnz = rows_end[m - 1];
+          using ptr_type = decltype(row_mapC.data());
+          using rmptr_type = typename std::remove_pointer<ptr_type>::type;
+          using nonconst_rmptr_type = typename std::remove_const<rmptr_type>::type;
```

Comment:
This can be simplified using the typedefs defined in Kokkos::View.
I suggest doing the following:
```c++
using non_const_size_type = typename cin_row_index_view_type::non_const_value_type
auto* tmpPtr = const_cast<non_const_size_type*>(row_mapC.data() )
```
or is there something I missed?

Reply:
@lucbv I originally did something like that, but then started overthinking with concern about what happens with things like FAD or ensemble types, is `non_const_value_type` the right type to use in all cases? I thought maybe `decltype` etc was a way to let the machine figure it out in each instance, but I'm not fully confident and happy to make the change to what you suggest if you prefer.

Reply:
This is the row_map so the `value_type` should be an integral type. Also on line 345 we throw if the values are not float or double so I think we are fine ; )

Reply:
> This is the row_map so the `value_type` should be an integral type

Ah, yep... good catch. Let me push the cleaner change.